### ISSUE TITLE
Bug: [no-unsafe-*] false positive for TS4.7's expression with type arguments

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#5065]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.1-rc&sourceType=module&showAST=es&code=PQKgBApgzgNglgOwC5gAJIJ4AdoGMBOcWSAtNPMsAgPYkCuCUAhgGYQlNRRwDmCAthGQAuMAG0A5BHz5q+CQF0wIYAFgAUBpYNcSONQRh+TANYQAQtQAeAHgAqAPgAUANyYw6EUXYCUYAN4aYGD4EEh0+Ib+YG4eEGAAvgDcGgkaGrgGUCjGZgDKSIQIPJZWYAC8RqYW1jbZRTwOSUA&rules=N4XyA&tsConfig=N4XyA

### Repro Code

```typescript
/* eslint @typescript-eslint/no-unsafe-assignment: ['error'] */

function makeBox<T>(value: T) {
  return { value };
}

const makeStringBox = makeBox<string>;
```

### ESLint Config

```javascript
module.exports = {
  "parser": "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/no-unsafe-assignment": ["error"]
  }
}
```

### Expected Result

No error

### Actual Result

```ts
const makeStringBox = makeBox<string>;
//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Unsafe assignment of an `any` value.
```

### Additional Info

This is a bug with TS - follow along here https://github.com/microsoft/TypeScript/issues/49223

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.26.0` |
| `@typescript-eslint/parser` | `5.26.0` |
| `TypeScript` | `4.7.1-rc` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
Closing as this is a bug with TS
https://github.com/microsoft/TypeScript/issues/49223
